### PR TITLE
Fix HTTP Auth

### PR DESCRIPTION
**Pull request by evancrow**
Caused by: #4195, I guess we still need the LoginHelper code.

### How was this tested?
- [x] Existing automated tests are enough to cover my changes

## Issues addressed
Fixed: #1755
Fixed: #1958
Fixed: #4346

### CODE REVIEW

File: Client/Frontend/Browser/ContentScripts/LoginsHelper.swift [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+import Foundation
+import Shared
+import Storage
+import SwiftyJSON
+import WebKit
+import XCGLogger
+
+private let log = Logger.browser
+
+class LoginsHelper: TabContentScript {
```
Comment (darinwf) [inline Client/Frontend/Browser/ContentScripts/LoginsHelper.swift:15]: I'm surprised we still need this system for HTTP auth.

**Review by darinwf — COMMENTED** [review]
I don't think we need the LoginsHelper part. The HTTP Auth system is just driven from a webview delegate method, right?

**Comment by evancrow** [comment]
That's what I originally thought, but `Authenticator.handleAuthRequest` isn't ever called by the `WebView`, but by `LoginHelper.requestLogin`.

**Comment by evancrow** [comment]
Wait, it doesn't look like it's called from there... I think I can remove it again.

**Review by darinwf — APPROVED** [review]

**Comment by darinwf** [comment]
Hopefully the tests will be stable / not flaky given the tweaks you made.

**Comment by evancrow** [comment]
Yeah I'll run them a few times to be sure.

**Comment by evancrow** [comment]
Marking as cherry-pick-needed since this fixes a crash.

**Pull request merged**